src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocketState.cs

```diff
@@ -37,9 +42,35 @@ internal sealed class WinHttpWebSocketState : IDisposable
 
         public WinHttpWebSocketState()
         {
+        }
+
+        public void Pin()
+        {
+            Debug.Assert(!_operationHandle.IsAllocated);
             _operationHandle = GCHandle.Alloc(this);
         }
 
+        public void Unpin()
+        {
+            if (_operationHandle.IsAllocated)
+            {
+                Debug.Assert(_handlesOpenWithCallback == 0);
+
+                _operationHandle.Free();
+                _operationHandle = default(GCHandle);
+            }
+        }
+
+        public int IncrementHandlesOpenWithCallback()
```

Comment:
Nit: should this return void?  Or do you expect a caller will end up needing this?